annotator-scanner/src/main/java/edu/ucr/cs/riple/scanner/Serializer.java

```diff
@@ -136,11 +157,42 @@ private void initializeOutputFiles(Config config) {
       if (config.classTrackerIsActive()) {
         initializeFile(classInfoPath, ClassInfo.header());
       }
+      initializeFile(nonnullInfoPath, SymbolLocation.header());
     } catch (IOException e) {
       throw new RuntimeException("Could not finish resetting serializer", e);
     }
   }
 
+  /**
+   * Converts the given uri to the real path. Note, in NullAway CI tests, source files exists in
+   * memory and there is no real path leading to those files. Instead, we just serialize the path
+   * from uri as the full paths are not checked in tests.
+   *
+   * @param uri Given uri.
+   * @return Real path for the give uri.
+   */
+  @Nullable
+  public static Path pathToSourceFileFromURI(@Nullable URI uri) {
```

Comment:
I reused classes from `Location` package in [here](https://github.com/uber/NullAway/tree/master/nullaway/src/main/java/com/uber/nullaway/fixserialization/location). In constructor of `AbstractSymbolLocation` in that package, there is a call to this method. Since I assumed I should not modify the copied source code from Uber, I added this method. Please let me know if I should follow a different approach. Thank you.

Reply:
Are you saying this function is not used at all in this project?  If so, I think it's best to delete the function

Reply:
Actually I see how that the function is being called.  But now I am wondering, why are all these classes from the `location` package getting copied over in this PR?  I thought this PR was about (1) serializing where `@Nonnull` annotations appear and then (2) if there is a `@Nonnull` annotation somewhere, don't allow there to be a fix that also adds `@Nullable` at that location.

Reply:
Yes, that is the intention of this PR. I was using Scanner to serialize the location of elements with explicit `@Nonnull` annotations which annotator-core can retrieve them to prevent from generating fixes for such elements. The serialization of elements in scanner is using the same API implemented in NullAway for serializing a location for an element. The other alternative is to use javaparser, but for each fix, we have to parse the source file and check for existing `@Nonnull` annotations during the complete execution of Annotator. But with this approach, we only index all once during the first build of target and reuse it during the process.  We may be able to speed up the javaparser approach with some caching techniques, but I still think the current implementation (indexing all at once) is faster. Please let me know which approach you prefer and I will update this PR, thank you.

Reply:
FYI, since `pathToSourceFileFromURI` will be added to Scanner, I made #143 as a follow up for this which uses the same API used in NullAway for all paths serializations. Will land that once this PR is landed.